src/c-writer.cc

```diff
@@ -3327,7 +3327,13 @@ void CWriter::WriteTryCatch(const TryExpr& tryexpr) {
   Write("const wasm_rt_tag_t ", tlabel, "_tag = wasm_rt_exception_tag();",
         Newline());
   Write("uint32_t ", tlabel, "_size = wasm_rt_exception_size();", Newline());
-  Write("void *", tlabel, " = alloca(", tlabel, "_size);", Newline());
+  // check size for exhaustion
+  // this is done here to allow the runtime to store exceptions in malloc, even
+  // if it does not currently do so. also, it is possible to compile different
+  // modules with different exception sizes and still link them together.
+  // (whether you'll run into issues will heavily depend on the use-case.)
+  Write("if (", tlabel, "_size > WASM_EXN_MAX_SIZE) { TRAP(EXHAUSTION); }");
```

Comment:
Can we perhaps assert this statically?

Reply:
the only reason we don't is that we'd prefer the ability to use different `WASM_EXN_MAX_SIZE` per module, until we get around to implementing GC

Reply:
But `WASM_EXN_MAX_SIZE` is not variable, its fixed in the header file in source control.   I don't think we need to worry about variable `WASM_EXN_MAX_SIZE` (at least not until we have to).

How about replacing this with `assert( tlabel, "_size <= WASM_EXN_MAX_SIZE)`?

Reply:
if we're gonna do that, we may as well go a little extra. so, how about this?